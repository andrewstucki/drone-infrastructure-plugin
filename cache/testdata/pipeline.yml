---
kind: pipeline
name: include

cache:
  - path: node_modules
    hash: yarn.lock
  - path: something
  - hash: foo # this should be ignored
steps:
  - name: build
    image: golang:1.11
    commands:
      - go get -u github.com/golang/dep/cmd/dep
      
---
kind: secret
type: external
data:
  password: 'drone/docker#password'
  username: 'drone/docker#username'
