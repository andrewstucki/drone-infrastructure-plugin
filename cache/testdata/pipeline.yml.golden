name: include
kind: pipeline
steps:
- image: andrewstucki/s3-cache
  name: Restore node_modules
  settings:
    access_key:
      from_secret: cache_access_key
    hash: yarn.lock
    pull: true
    restore: true
    root:
      from_secret: cache_bucket
    secret_key:
      from_secret: cache_secret_key
- image: andrewstucki/s3-cache
  name: Restore something
  settings:
    access_key:
      from_secret: cache_access_key
    hash: ""
    pull: true
    restore: true
    root:
      from_secret: cache_bucket
    secret_key:
      from_secret: cache_secret_key
- commands:
  - go get -u github.com/golang/dep/cmd/dep
  image: golang:1.11
  name: build
- image: andrewstucki/s3-cache
  name: Uploading node_modules
  settings:
    access_key:
      from_secret: cache_access_key
    hash: yarn.lock
    mount:
    - node_modules
    pull: true
    rebuild: true
    root:
      from_secret: cache_bucket
    secret_key:
      from_secret: cache_secret_key
- image: andrewstucki/s3-cache
  name: Setting TTL for node_modules
  settings:
    access_key:
      from_secret: cache_access_key
    flush: true
    flush_age: 5
    hash: yarn.lock
    pull: true
    root:
      from_secret: cache_bucket
    secret_key:
      from_secret: cache_secret_key
- image: andrewstucki/s3-cache
  name: Uploading something
  settings:
    access_key:
      from_secret: cache_access_key
    hash: ""
    mount:
    - something
    pull: true
    rebuild: true
    root:
      from_secret: cache_bucket
    secret_key:
      from_secret: cache_secret_key
- image: andrewstucki/s3-cache
  name: Setting TTL for something
  settings:
    access_key:
      from_secret: cache_access_key
    flush: true
    flush_age: 5
    hash: ""
    pull: true
    root:
      from_secret: cache_bucket
    secret_key:
      from_secret: cache_secret_key
---
name: ""
kind: secret
data:
  password: drone/docker#password
  username: drone/docker#username
type: external
---
name: cache_access_key
kind: secret
get:
  name: cache_access_key
  path: drone/management-access-key
---
name: cache_secret_key
kind: secret
get:
  name: cache_secret_key
  path: drone/management-secret-key
---
name: cache_bucket
kind: secret
get:
  name: cache_bucket
  path: drone/cache-bucket
